### PR TITLE
Update Move examples of EC operations

### PR DESCRIPTION
**Pull request by benr-ml**
## Description 

Updated the examples from `sui_programmability/examples/crypto/sources/ec_ops.move`.

## Test plan 

How did you test the new or updated feature?

---

## Release notes

Check each box that your changes affect. If none of the boxes relate to your changes, release notes aren't required.

For each box you select, include information after the relevant heading that describes the impact of your changes that a user might notice and any actions they must take to implement updates. 

- [ ] Protocol: 
- [ ] Nodes (Validators and Full nodes): 
- [ ] Indexer: 
- [ ] JSON-RPC: 
- [ ] GraphQL: 
- [ ] CLI: 
- [ ] Rust SDK:

### CODE REVIEW

File: examples/move/crypto/elgamal/sources/example.move [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+        to_hash.append(*bytes(&enc1.ephemeral));
+        to_hash.append(*bytes(&enc1.ciphertext));
+        to_hash.append(*bytes(&enc2.ephemeral));
+        to_hash.append(*bytes(&enc2.ciphertext));
+        to_hash.append(*bytes(a1));
+        to_hash.append(*bytes(a2));
+        to_hash.append(*bytes(a3));
+        let mut hash = blake2b256(&to_hash);
+        // Make sure we are in the right field. Note that for security we only need the lower 128 bits.
+        *vector::borrow_mut(&mut hash, 0) = 0;
```
Comment (amnn) [inline examples/move/crypto/elgamal/sources/example.move:52]: I'm guessing that this doesn't work yet, right?

```suggestion
        hash[0] = 0;
```

File: examples/move/crypto/ibe/sources/example.move [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+
+        IbeEncryption { u, v, w }
+    }
+
+    #[test_only]
+    /// Encrypt a message 'm' for 'target'. Follows the algorithms of https://eprint.iacr.org/2023/189.pdf.
+    /// Note that the algorithms in that paper use G2 for signatures, where the drand chain uses G1, thus
+    /// the operations below are slightly different.
+    public fun insecure_ibe_encrypt(pk: &Element<G2>, target: &vector<u8>, m: &vector<u8>, sigma: &vector<u8>): IbeEncryption {
+        assert!(vector::length(sigma) == 32, 0);
```
Comment (amnn) [inline examples/move/crypto/ibe/sources/example.move:48]: `sigma.length()` (couple more examples below).

**Review by amnn — APPROVED** [review]
Thank you!

File: examples/move/crypto/elgamal/sources/example.move (hunk 1 above)
Comment (benr-ml) [reply examples/move/crypto/elgamal/sources/example.move:52]: yes, left-hand positions not supported yet

**Pull request merged**